app/controllers/foreman_ansible/api/v2/hostgroups_controller_extensions.rb

```diff
@@ -5,25 +5,19 @@ module V2
       module HostgroupsControllerExtensions
         extend ActiveSupport::Concern
         include ForemanTasks::Triggers
+        include ::ForemanAnsible::Concerns::JobInvocationHelper
 
         # Included blocks shouldn't be bound by length, as otherwise concerns
         # cannot extend the method properly.
-        # rubocop:disable BlockLength
         included do
           api :POST, '/hostgroups/:id/play_roles',
               N_('Plays Ansible roles on a hostgroup')
           param :id, String, :required => true
 
           def play_roles
             find_resource
-
-            @result = {
-              :hostgroup => @hostgroup, :foreman_tasks => async_task(
-                ::Actions::ForemanAnsible::PlayHostgroupRoles, @hostgroup
-              )
-            }
-
-            render_message @result
+            composer = job_composer(:ansible_run_host, @hostgroup.hosts)
```

Comment:
`@hostgroup.hosts` only links direct assignments ignoring the hostgroup inheritance, perhaps this is what we want but wanted to raise it here

Reply:
Good point, will address on the Hostgroup PR - I think it makes more sense to run the whole tree